### ISSUE TITLE
Support multicast dns for name and service resolution

### ISSUE THREAD

**Issue by kevinanderson1** [tailscale/tailscale#1013]
Tailscale makes it super easy to build and maintain a mesh network with Wireguard. It would be nice if multicast DNS support could be added to make DNS name and service discovery easier for connected peers. 

Magic DNS doesn't necessarily fill this gap in my use case as mdns also provides service discovery for things like airprint.

I've been able to get multicast dns working with "native" wireguard by manually enabling multicast on the interface and setting the allowed IP's to 0.0.0.0/0. 

**Describe the solution you'd like**

Support multicast traffic across tailscale specifically for multicast dns.

**Describe alternatives you've considered**

I have setup native wireguard, enabled multicast on the interfaces, and set the allowed IP's to 0.0.0.0/0 but tailscale makes the management easier for situations where certain endpoints are behind dynamic IP's.

**Comment by normanr** [comment]
Should this work with advertised routes too? i.e. also proxy the multicast traffic from devices on advertised routes to devices that have accepted those routes?

**Comment by kevinanderson1** [comment]
My initial thought is that would be ideal. My use case is multiple nodes connecting to my home router. Within my LAN I am advertising multiple services through mDNS that I would like to access from clients.

**Comment by normanr** [comment]
I noticed that avahi-daemon has a reflector section. I haven't tried it, but I assume it won't work on its own because tailscaled won't route the mDNS traffic.

**Comment by kevinanderson1** [comment]
Yes that is what I did on a native wireguard connection which was to enable
reflection on that interface so it would take mDNS packets from the LAN
side and inject them to the wireguard interface but as you stated that
won't work since tailscale does not seem to be passing the mDNS traffic.

**Comment by normanr** [comment]
I assume setting up a GRE tunnel (using the tailscale IP addresses), and routing multicast traffic over it (with something like smcroute) would also work. That would allow you to use Tailscale to manage the network, and get the multicast packets through the Tailscale network.

**Comment by pranavpeshwe** [comment]
Hello! Any news/development on the mDNS front? 
Tailscale is (obviously!) very useful. Making mDNS work on it seamlessly has the potential to make Tailscale indispensable.

Thanks and regards.

**Comment by DentonGentry** [comment]
We're tracking this issue in the engineering backlog and will post updates as progress is made.

**Comment by bentumbler** [comment]
+1 for mDNS and bonjour.
My use case is to capture debug logs from Apple devices as I roam between Wi-Fi and Cell

**Comment by alie2n** [comment]
+1 for mDNS and bonjour
My use case is more mundane. I just want to be able to print on the go. :)

**Comment by sylr** [comment]
Time machine backups over Tailscale 🤤

**Comment by brutella** [comment]
My use case would be to get remote access to HomeKit accessories.
HomeKit accessories announce IP addresses using mDNS on the local network.

**Comment by ties** [comment]
> Time machine backups over Tailscale 🤤

To clarify: This is currently blocked because it seems to require mDNS.

**Comment by movinimage** [comment]
Would like to use for:

Time Machine
AirPrint
Homekit

all require Bonjour / mDNS

**Comment by aikinai** [comment]
I've been Googling this every few years for decades to see if anyone has solved it yet, so very excited to find this thread and see there's some traction! Like others, I would also use it for HomeKit, Time Machine, Home Media Sharing, etc.

Really looking forward to it!! ٩(◕‿◕｡)۶

**Comment by BasixKOR** [comment]
It seems like [Google Cast also uses mDNS](https://github.com/hensm/fx_cast/blob/master/app/src/bridge/components/discovery.ts#L35) to discover other Chromecast in the network, so this could (theoretically) allow the Tailscale nodes to cast to connected Chromecast built-in Android TVs.

**Comment by dsemaya** [comment]
I just started using Tailscale. I was super excited since it seems like it should be a worthy replacement to Back To My Mac that finally works with iOS…except the lack of Bonjour resolution. 

BTMM did this without multicast. It DNS-SD, but in a private way where you were the only one who could resolve your own hosts, such as hostname.username.members.mac.com. But you rarely needed to remember these names since they just showed up everywhere: in the Finder for file sharing and screen sharing, and even in Terminal for SSH connections! 

I think this write up should help. https://datatracker.ietf.org/doc/html/rfc6281#section-1

**Comment by dvtkrlbs** [comment]
What I wonder is does the Multicast Entitlement requirement for IOS prevents this feature. Or is Tailscale eligible for this entitlement ?

**Comment by tysecure** [comment]
Is this why I cannot print to my Bonjour printer even though File and Screen sharing access works flawlessly? 🙈 I've been on this for an hour before stumbling upon this thread 😅

**Comment by EpsilonAlpha** [comment]
+1 for mDNS Routing

I'm having a homebridge at home and the scenes are now bugged on all Apple devices as soon I turned on tailscale. When i switch devices directly it works but i need something like "i'm at home, completely"

**Comment by tiehfood** [comment]
+1 needed for music streaming

**Comment by asdffdsazqqq** [comment]
> My usecase: let vlc on my android tablet see the DLNA server on the home network when I'm away

hi, thanks, did you every find a solution, as i am in of the same thing?
running `rclone serve dlna` on a hetzer vm and vlc on android.
the thing is that running `rclone serve sftp`, vlc can find the sftp server.
not sure but i assume the reason is sftp uses tcp and dlna uses udp

**Comment by jankatins** [comment]
> hi, thanks, did you every find a solution, as i am in of the same thing?

Nope, did something similar: used a file share (which went to the IP) to pull the videos to the tablet and watched it from local.

**Comment by eladrave** [comment]
TailsScale works in layer 3+ of the OSI layers. The virtual NiC is a TUN device. This is obviously better for compatibility.
In order to work with mDNS, the vnic needs to be a TAP device, which cannot be done on iPhone.

There are some ways to solve it (like have th exit point "proxy" the layer 2 protocol, e.g. ARP etc...).

ZeroTier is a layer 2 service (virtual Ethernet), so that's why it works. 

A couple of years ago I managed to get a layer 3 work with layer 2 by creating 2 vnics on the exit node one for the VPN (layer3) and the 2 one was the OpenVPN vnic that can be configured at TAP (just the vnic, not the full software), and having the TAP device be the "gateway" for the exit node.
(This was not done with TailScale, but I guess this could work here also)

**Comment by dm3ch** [comment]
If we are discussing mDNS itself for me it seems easier to just implement mDNS proxying instead of making L2 VPN over L3 VPN :)

**Comment by asdffdsazqqq** [comment]
@dm3ch, hello, thanks, interesting suggestion

i am running the rclone dlna server on ubuntu.
please, to share a how-to guide or internet link about how to implement mDNS proxy

**Comment by dm3ch** [comment]
Unfortunately, I have no working setup.

I just found that some of the firewalls like pfSense have an mDNS repeater/proxy feature that allows mDNS discovery works in neighborhood networks. So I assume that something like this could be implemented in Tailscale.

But the difference between a firewall and Tailscale is that firewalls have access to both L2 networks from the same machine, so the repeater, as far as I understood, just receives a message in one L2 segment and sends it to another L2 segment. But in the case of tailscale interfaces on clients are L3, so as far as I understand even if that's possible it would require MDNS proxy to be set up on each client and somehow communicating with each other.

So my idea is something like this https://github.com/wisq/mdns-proxy but implemented inside tailscale client and tight integrated with it.

**Comment by MrBuddyCasino** [comment]
This would also be useful for smart-home stuff like Home Assistant that is based on mDNS auto-discovery in a local network. A sensor in another building could easily be added transparently, when Tailscale is installed on the router.

**Comment by j-f1** [comment]
Would it be possible to implement a simpler version by taking advantage of macOS’s “Use dynamic global hostname” setting to have the system push the appropriate DNS records as they change?

**Comment by footlooseboss** [comment]
> “Use dynamic global hostname” setting 

https://apple.stackexchange.com/a/1035

TL;DR: a type of DDNS client requiring a provider that supported wide area bonjour -  that allow(ed) macOS server / AirPort router to publish /set Bonjour/mDNS advertisements/records on the DDNS address, then by adding it to search domains, stuff like airprint and home sharing is ..widely available 

I know it was discussed in the forums a while ago and faced barriers probably similar to above.



https://datatracker.ietf.org/doc/rfc8766/)

**Comment by renne** [comment]
For embedded devices which do not support mDNS I suggest to consider proxying ARP and NDP or insert learned ARP/NDP records into MagicDNS. This would create a hell of an overlay network via Tailscale. :-)

**Comment by adipierro** [comment]
Unicast DNS-SD would be nice too – right now Tailscale does not add them to recommended browsing domains on macOS (can be seen with `dns-sd -F`), and removes all locally-added ones.

**Comment by bobbyl140** [comment]
Adding my two cents. I have 3 servers, one of which is behind NAT with DMZ setup. This prevents UPnP port mappings from other LAN devices. The only reason I need UPnP port mappings is to make unicast DNS-SD word with a BIND name server. If I could register Tailscale IPs somehow that would make Tailscale feel like magic even more.

**Comment by LegendusMaximus** [comment]
As others have said, it would be amazing if Tailscale could support m'DNS.
Possible use cases include administering my AirPort Express remotely (AirPort utility needs mDNS for device discovery), remote printing via iOS AirPrint, remote AirPlay and much more.
It would make Tailscale seem even more magic if this could work seamlessly like the rest of Tailscale already does.

**Comment by dend** [comment]
I'll throw my name in the hat for supporting this feature. Specifically, I think this would help enable UniFi applications (network and their camera management Protect tooling) to work seamlessly over Tailscale - it's currently utterly broken because it relies on in-network discovery.

**Comment by woferry** [comment]
+1 for me.  I was hoping to use Tailscale on an AppleTV at my parents house to access the media I have shared via Home Sharing on a Mac Mini at my house.  But it seems this isn't possible today and would require this feature.

**Comment by vassoz** [comment]
+1

**Comment by danielraffel** [comment]
Also wish this support existed to access some smart home devices outside the local network.

**Comment by roisec** [comment]
+1

**Comment by cd-slash** [comment]
Found this thread while looking for solutions to let me back up my MacBook to my Google Drive. I have a Digital Ocean Droplet running various Docker containers, one of which is [Docker Timemachine](https://github.com/mbentley/docker-timemachine).

I used this successfully before in a homelab environment and couldn't work out why it wouldn't work here when everything else works... but it's because the mDNS doesn't work with Tailscale, so the Bonjour / Avahi resolution doesn't let it see the timemachine. Looking into mDNS proxy options, but a tailscale solution would be amazing as it means I could keep my laptop backed up while travelling, with a fast and low-latency connection to Digital Ocean 👍

**Comment by codyc1515** [comment]
I had totally forgotten about that. That would be a blocker for moving to use Tailscale Magic DNS for the whole OS.

**Comment by WhyNotHugo** [comment]
mDNS is used by device on the same network to discover each other.

This limitation kills the entire value of tailscale for me; I can have multiple of my devices around the work in the same virtual network, but they cannot discover each other using the standard techniques used in every other context.

It's really hard to understand how this has not been amongst the highest priorities. Devices being able to discover each other in the usual ways over the network is a key functionality of a network.

**Comment by synthetc82** [comment]
Just adding my interest for this as well, this is one of the things stopping us using tailscale in our corp environment, as we need to "cast" to office TV displays (don't ask not my idea) from HO.

**Comment by Ghxst** [comment]
+1 for this, would be very nice to have mDNS work on L3 without proxying and open up a lot of use cases me personally as well as within our company.

**Comment by craftcm** [comment]
Adding my support for this request. I would be a gamechanger for me to be able to use TailScale to remotely backup my iPhone via iMazing whenever I'm on the road away from home.

**Comment by WesleyBryie** [comment]
+1 pymobiledevice3 detection over tailscale would be great for projects like SideStore

**Comment by t4ec** [comment]
Was trying to do mDNS for my homekit cameras, streaming is broken from one site to another due to this limitation. I have configured mDNS reflectors but it seems the tailscale is the issue. So really thinking of removing tailscale and ditching it for simple wireguard setup. This actually a must-have feature since a lot of smart home hardware requires it to work.

**Comment by peskyAdmin** [comment]
> My use case would be to get remote access to HomeKit accessories. HomeKit accessories announce IP addresses using mDNS on the local network.

+ 1 using home-assistant + TS sidecare i am unable to discover homekit devices due to mDNS block

**Comment by sudocarlos** [comment]
+1 I'm hoping to access some services made available via .local addresses. For now im using ssh tunnels but it could be so much cleaner.

**Comment by zez3** [comment]
Just as info, since I stumbled yesterday on a similar issue with a different VPN client.

On MacOS and perhaps also iOS, Apple chose to ignore the `utun` (point 2 point) interfaces in their mdns bonjour implementation

Here is the piece of code from Apple itself:
https://github.com/apple-oss-distributions/mDNSResponder/blob/8f70f98fc1d0cf439ca3a6470be6ad8ac2bcc019/mDNSMacOSX/mDNSMacOSX.c#L192 
```
// We only attempt to send and receive multicast packets on interfaces that are
// (a) flagged as multicast-capable
// (b) *not* flagged as point-to-point (e.g. modem)
```

So the builtin Airplay and other similar services depending on mdns(bonjour) will fail to work/discover even if the broadcast and multicast is arriving on the VPN interface.

I am now wondering if I could somehow flag the tun interface as multicast-capable.

**Comment by andrekaufman** [comment]
+1 I'd like to use TailScale to remotely back up my iPhone via iMazing while not at home.

**Comment by coltography** [comment]
+1 I really needs mDNS to work with tailscale. would be so nice.

**Comment by xurtis** [comment]
@zez3 looks like there's an entitlement for multicast: [`com.apple.developer.networking-multicast`](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_networking_multicast). It looks like the developer account needs to be explicitly approved to use the entitlement.

**Comment by WesleyBryie** [comment]
> @zez3 looks like there's an entitlement for multicast: [`com.apple.developer.networking-multicast`](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_networking_multicast). 

But will this entitlement apply to VPN connections?

**Comment by zez3** [comment]
> @zez3 looks like there's an entitlement for multicast: [`com.apple.developer.networking-multicast`](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_networking_multicast). It looks like the developer account needs to be explicitly approved to use the entitlement.

My information was related to macos implementation of mDNSResponder/bonjour service. Not the VPN interface. I know and have tested the multicast over vpn with a different solution. THAT WORKS. The multicast reaches the macos network stack. I can see that in wireshark/tcprump. The issue is that this multicast traffic is simply ignored by the local macos built-in dns service that would use this for specific services(aka Airplay, etc.)

If you need the multicast over vpn for other Applications that do not use/depend on the built-in dnsResponder/resolver then it should work.

**Comment by fcassata** [comment]
Any news on this?

**Comment by polymo1** [comment]
> Any news on this?

If there was you would've heard it

**Comment by yqs112358** [comment]
Anyway, there should be a simple way for developers to **iterate over** all devices in the Tailscale network, and communicate with multiple nodes if needed. 🤔

For example, if I want to create a "universal clipboard" between devices in a Tailscale network, it's not very elegant to manually configure a "server", as this requires a particular node to be online all the time. Support for UDP Multicast (or merely mDNS) makes sense in this case, with which we can use common protocols such as Bonjour for device discovery and broadcast communication.

Theoretically, tsnet's `*LocalClient.Status` API is also able to help iterating over all devices in Tailscale network, but it may introduce some instability internal API or structs in Tailscale. In addition, this interface is not designed for this purpose.

**Comment by NireBryce** [comment]
throwing another hat into the ring that I'd like this too, as many selfhosted solutions like runtipi and (I think) HomeAssistantOS run their own mdns that seems to make my tailscale choke.

has anyone tried if `tailscale set --accept-dns=false` is the workaround here?  I'm headed to bed but I half remembered the [knowledgebase article about resolv.conf overwriting](https://tailscale.com/kb/1235/resolv-conf).

> If a DNS manager isn't available for your system, or you don't want to run one, and don't want Tailscale to overwrite /etc/resolv.conf, you can either disable MagicDNS for all devices on your tailnet or run `tailscale set --accept-dns=false` to disable MagicDNS on a single device.
> 
> Even if you set `--accept-dns=false`, Tailscale's MagicDNS server still replies at 100.100.100.100 (or fd7a:115c:a1e0::53), as long as MagicDNS is enabled on the tailnet. If you'd like to manually configure your DNS configuration, you can point *.ts.net queries at 100.100.100.100. The [100.100.100.100](https://tailscale.com/kb/1381/what-is-quad100) resolver runs inside tailscaled on the device and replies authoritatively to Tailscale DNS names without needing to forward queries out to the network.

**Comment by alx-xlx** [comment]
I don't think this will be solved seeing how long this issue has been up. The only possible reason I see is probably no profit model for tailscale to actually work on this. There is zerotier that works perfectly fine.

However, I just hope if there is anyway to piggyback on https, tcp, or udp to get this working ? I would like to be able to print my stuff over the network. isn't there a service where I just upload the pdf to the remote device and it will automatically print it.

**Comment by ericukk** [comment]
I wonder why this thread smells so Apple. Do other platforms not use mDNS? 
I'm new to networking and trying to use mDNS for my own network setup. But it seems that's not what people usually use it for.

**Comment by jfroy** [comment]
mDNS is a key component of Thread and Matter. If you somehow wanted to span a Matter fabric over Tailscale, you'd need it.

**Comment by lorenzo-deluca** [comment]
Any news??! Thanks a lot

**Comment by JayBrown** [comment]
+1

One of my use cases is controlling a macOS home server with (shell) scripts using Mousedown's iOS app **[EventScripts Mobile](https://www.mousedown.net/software/EventScriptsMobile.html)**, which connects via Bonjour, i.e. on the local network, to another Mac that's running the **[main EventScripts app](https://www.mousedown.net/software/EventScripts.html)** (with EventScripts server functionality enabled).

Nota bene: there's also the app **[Shelley](https://github.com/tylerhall/Shelley)** by @tylerhall … so it might also be possible to cobble together an alternative using `curl` and port 9876 via tailnet. (❓)

**Comment by peskyAdmin** [comment]
FYI there are fundamental challenges with mDNS. first challenge is that mDNS spans layer2 and layer3 of the OSI model. tailscale only operates at layer 3. this is the main problem. there is another additional apple related problem that apple products only conduct mDNS (due to the layer2 usage) to the interface where the LAN is instead of the tun interface made by wireguard/tailscale. some sort of relay needs to be built that listens for mDNS traffic encapsulates the layer2 initiation of the protocol over layer 3. nodes which have mDNS relay enabled can then deencapsulate the packet and send it on layer2. this COULD allow mdns to work. it also i a major lift and probably not being proritized as this is most likely not something enterprises are looking for, only us home labers

**Comment by xHyperElectric** [comment]
Multicast would be so useful to support playing old school PC games via LAN on tailscale. Then Tailscale would be a full replacement for Hamachi (and much much much better might I add). As well as adding printers remotely.

**Comment by chwiese** [comment]
For anybody wanting to run MacOS Time Machine over your tailnet:
You can set the destination with the `tmutil` command line utility like so `sudo tmutil setdestination smb://user@host/share`. Add the `-p` switch if your share has a password. See `tmutil --help` for more options, including setting a quota. Using this method, no mdns is required (but it seems a good idea to enable Magic DNS and use that as the host name for the destination).

**Comment by kaovilai** [comment]
Usecase is https://rogueamoeba.com/airfoil/ over tailscale while on isolated public wifi networks.

**Comment by javedbartlett** [comment]
Folks, it’s been half a decade. I don’t think the Tailscale team are going to add mDNS over Tailnet. We’d need to switch to a different product that supports this sadly.

**Comment by Ghxst** [comment]
> Folks, it’s been half a decade. I don’t think the Tailscale team are going to add mDNS over Tailnet. We’d need to switch to a different product that supports this sadly.

Feel free to share anything you find that supports this.

**Comment by limoner0** [comment]
+1. If you use Avahi with specific local domains, it's a pain...

**Comment by Nachlor** [comment]
> Multicast would be so useful to support playing old school PC games via LAN on tailscale. Then Tailscale would be a full replacement for Hamachi (and much much much better might I add). As well as adding printers remotely.

You can use Broadcast / Multicast by routing Tailscale through Lanemu P2P VPN:
- https://gitlab.com/Monsterovich/lanemu

Check this short guide:
- https://pastebin.com/raw/htmb6PVu

Instead of port forwarding, simply specify your Tailscale Machine IP (e.g., 100.111.88.108:5521) in Known Hosts when creating a network in Lanemu :)

**Comment by John-Colvin** [comment]
Supporting this could reduce a significant friction - really enable a whole new way of working - for developing ios apps remotely. I would pay.

**Comment by fogwandering** [comment]
We're a small company looking at various ways to get off our legacy VPNs, and we'd probably become customers if we could allow users to access their printers and SMB shares without manually setting IPs per Tailscale.

**Comment by yuhenghuang** [comment]
> mDNS support is one of the few areas where ZeroTier is clearly superior to Tailscale.

Has anyone tried using ZeroTier for mDNS support along side with Tailscale as the main VPN?

In the settings where there are two routers, ideally linux ones, installing Zerotier on two routers and configuring avahi reflectors should do the job.

**Comment by yqs112358** [comment]
The question is: When we need to install ZeroTier extra to realize something, why not just directly replace Tailscale with ZeroTier? 🤣

This is Tailscale's biggest problem at the moment: Missing important features leads to a loss of current users

**Comment by lel-amri** [comment]
Hello dear maintainers,
Could someone close this issue to prevent non-contributors from posting?
I've been subscribed for a couple of years; except for rare occurrences of insightful information, every comment can be classified as spam.
The problem is very well known and delimited already, I don't think anyone can add something meaningful to the description of the issue anymore. People who want the feature can just post a thumb up to the initial comment. I prefer to get notified only when either someone start contributing something, or the maintainers have decided a stance regarding this issue.

**Comment by DimitarNestorov** [comment]
> Could someone close this issue to prevent non-contributors from posting?

@lel-amri you can change your notification settings for the issue.

<img width="514" height="457" alt="Image" src="https://github.com/user-attachments/assets/f3265312-2eca-4618-a42d-6aa39a46520f" />

> I don't think anyone can add something meaningful to the description of the issue anymore.

Limiting to contributions only will prevent non-contributors to share any new workarounds/solutions they may find in the future. Also non-contributors can suggest potential libraries/existing open source solutions the Tailscale contributors can use to solve this problem.

**Comment by lel-amri** [comment]
> > Could someone close this issue to prevent non-contributors from posting?
> 
> [@lel-amri](https://github.com/lel-amri) you can change your notification settings for the issue.

I still want to be notified about new comments that are actually related to solving this issue _in_ Tailscale.

> > I don't think anyone can add something meaningful to the description of the issue anymore.
> 
> Limiting to contributions only will prevent non-contributors to share any new workarounds/solutions they may find in the future. Also non-contributors can suggest potential libraries/existing open source solutions the Tailscale contributors can use to solve this problem.

That's the idea actually. In my opinion, workarounds and solutions (not related to fixing the issue in the software) should be discussed somewhere else than in the forge's issue system.

I think everything that could be discussed has been discussed: Tailscale (WireGuard, actually) isn't capable of forwarding/accepting packets to/from specific subnets (that aren't the peers' subnet) for more than one peer, so the only fix is for Tailscale (or any other software, really, some have already been discussed here) to forward the packets explicitly (which seems forbidden on some Apple devices).

My problem with this issue is that people just comment to say they want the feature (instead of using the thumb up emoji on the first comment). Some even try to jab at Tailscale maintainers with some passive-aggressive comments (in the spirit of "I won't use Tailscale because it does not support such a basic feature" or "Tailscale will lost all of their customers because they are unable to develop something so simple").

I re-iterate that it would benefit every subscriber's mailbox to lock this issue to non-commiters, such that we get notified only when work is made towards fixing this issue, or if maintainers wants some feedback from the community.

This is my last comment about closing this issue to non-commiters. Sending two comments about this is already too much. I'm sorry for the noise, subscribers.

**Comment by oguzhnatly** [comment]
+1 for mDNS Routing

**Comment by kevin-stuart** [comment]
A final thought by me: I think issues are useful for this to get a mail if something changes for this. Is there something else I can subscribe to get an email if something changes? Like a dicussion thread that one can subscribe to? But I do agree that this is not cleanly split here for code issues. Prople want different things.

**Comment by webwarrior06** [comment]
> Supporting this could reduce a significant friction - really enable a whole new way of working - for developing ios apps remotely. I would pay.

@John-Colvin I created a tool that makes possible to debug xcode apps remotely over any mesh VPN, including Tailscale. You need to pair the device once and connect remotely afterwards anywhere. 
Here is the MacOS app: https://nuticast.com

**Comment by damajor** [comment]
> > Supporting this could reduce a significant friction - really enable a whole new way of working - for developing ios apps remotely. I would pay.
> 
> [@John-Colvin](https://github.com/John-Colvin) I created a tool that makes possible to debug xcode apps remotely over any mesh VPN, including Tailscale. You need to pair the device once and connect remotely afterwards anywhere. Here is the MacOS app: https://nuticast.com

This issue is not about promotion on subscription of closed source software  !
